ledfx/effects/random.py

```diff
@@ -0,0 +1,92 @@
+import random
+import time
+
+import numpy as np
+import voluptuous as vol
+
+from ledfx.color import parse_color, validate_color
+from ledfx.effects.temporal import TemporalEffect
+
+
+class RandomEffect(TemporalEffect):
+    """
+    Randomly activates sections
+    """
+
+    NAME = "Random"
+    CATEGORY = "Non-Reactive"
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional(
+                "hit_color", description="Hit color", default="#FFFFFF"
+            ): validate_color,
+            vol.Optional(
+                "hit_duration",
+                description="Hit duration",
+                default=0.1,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.1, max=5.0)),
+            vol.Optional(
+                "hit_probability_per_sec",
+                description="Probability of hit per second",
+                default=0.1,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.01, max=1.0)),
+            vol.Optional(
+                "hit_relative_size",
+                description="Hit size relative to LED strip",
+                default=0.2,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.1, max=1.0)),
+        }
+    )
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+        self.seconds_since_last_hit = 0
+
+    def config_updated(self, config):
+        self.hit_color = np.array(
+            parse_color(self._config["hit_color"]), dtype=float
+        )
+        self.hit_relative_size = self._config["hit_relative_size"]
+        self.hit_duration = self._config["hit_duration"]
+        self.speed = self._config["speed"]
+        self.probability_per_sec = (
+            self.__balance_hit_probability_based_on_speed()
+        )
+
+    def on_activate(self, pixel_count):
+        self.seconds_since_last_hit = 0
+
+    def effect_loop(self):
+        hit_absolute_size = int(self.pixel_count * self.hit_relative_size)
+        self.__update_duration_since_last_hit()
+
+        hit_is_still_active = self.seconds_since_last_hit < self.hit_duration
```

Comment:
This should just be the if statement clause, doesn't need to be substantiated as a variable.

if not ( self.seconds_since_last_hit < self.hit_duration)

if similar

Reply:
I thought it would add to the readability, although it might be obsolete

Reply:
It makes you expect to see that used as a variable elsewhere, better to have a comment above to explain the logic if it seems hard to read. This is not a complex comparison, but comments don't hurt.